### ISSUE TITLE
Reuse existing GuzzleHTTP clients

### ISSUE THREAD

**Issue by andhikanugraha** [Azure/azure-storage-php#56]
I had a use case for creating many block blobs (65000+) at once to cache records in a database. However, the library would consistently throw an exception after the 2000th createBlockBlob call:

```
cURL error 7: Failed to connect to ***.blob.core.windows.net port 443: Address already in use (see http://curl.haxx.se/libcurl/c/libcurl-errors.html)
```

While troubleshooting, I found that `ServiceRestProxy` (inherited by `BlobRestProxy`) instantiates a new `GuzzleHttp\Client` on every call, whereas examples in the [Guzzle documentation
](http://docs.guzzlephp.org/en/latest/quickstart.html) reuse a single Client object across multiple requests. It seems that each `Client` object uses up its own outbound port, eventually exhausting the available ports on the server.

I was able to fix the issue by modifying `ServiceRestProxy->sendAsync` to reuse the `Client` object. I created a method in ServiceRestProxy that checks whether a `GuzzleHttp\Client` with provided `$clientOptions` has been created before. If not, the method would call `ServiceRestProxy->createClient` and return the resulting object.

I've submitted a PR at #57.

**Comment by katmsft** [comment]
@andhikanugraha Please check offline discussion email since we have not heard back from you for some time.

**Comment by katmsft** [comment]
@andhikanugraha This issue is resolved in release 0.15.0.
